FHEM/90_SIGNALduino_un.pm

```diff
@@ -281,18 +185,82 @@ SIGNALduino_un_Parse($$)
 		#Dispatch($hash, $tscode,undef);
 
 		
-	} else {
-		Log3 $hash, 4, $dummyreturnvalue;
+	}	
+								   
+	##############################################################################################
+	# version 1) message with u..# without development y attribut -> Unknown code u..# , help me!
+	# version 2) message with u..# and development y attribut -> no message for Unknown code
+	##############################################################################################
+	
+	my $value = AttrVal($ioname, "development", "none");
+	my @delevopmentargs = split (",",$value);
+	my $search = "y".$protocol;			# for version 2
+	my $search2 = "u".$protocol;		# for version 1
+
+	Log3 $name, 5, "$name: $ioname AttrVal development: $value" if (defined $value);
+	
+	if (grep{/$search$/ || /$search2$/}@delevopmentargs) {
+		### Help Device + Logfile ###
+	
+		Log3 $name, 4, "$name: $ioname Protocol $protocol ($search) found in AttrVal development!" if (grep{/$search$/}@delevopmentargs);
+		Log3 $name, 4, "$name: $ioname Protocol $protocol ($search2) found in AttrVal development!" if (grep{/$search2$/}@delevopmentargs);
+		
+		my $def;
+		my $devicedef = $name."_".$protocol;
+		$def = $modules{SIGNALduino_un}{defptr}{$devicedef} if(!$def);
+		$modules{SIGNALduino_un}{defptr}{ioname} = $ioname;
+		
+		if(!$def) {
+			Log3 $ioname, 1, "$ioname: $name UNDEFINED sensor " . $devicedef . " detected";
+			return "UNDEFINED $devicedef SIGNALduino_un $devicedef";
+		}
+	
+		my $hash = $def;
+		my $name = $hash->{NAME};
+		my $info = "$bitData | bits=".length($bitData);
+	
+		readingsBeginUpdate($hash);
+		readingsBulkUpdate($hash, "state", "newMSG - ".TimeNow(),0);
+		readingsBulkUpdate($hash, "LogInfo", $info)  if (defined($info));
+		readingsEndUpdate($hash, 1); 		# Notify is done by Dispatch
```

Comment:
wo ist denn $info definiet / wie kann ich es als Anwender beeinflussen. 
Ich habe wohl Tomaten auf den Augen :(

Reply:
@sidey79 
$info ist weiter oben definiert.
>wie kann ich es als Anwender beeinflussen.

Der Anwender kann es nicht beeinflussen. Darin werden die Informationen der DMSG gespeichert in Form der kompletten Bits binär und die Anzahl der Bits. Einstellbar würde ich da nichts machen weil so haben wir immer ständig die komplette DMSG + BITS binär und die Anzahl der Bits

Reply:
Okay, manchmal muss man sich den Code besser komplett ansehen.

In dieser Zeile wird $info ja mit einem Wert belegt. Folglich, müsste $info dann **immer** definiert sein. Richtig?
https://github.com/RFD-FHEM/RFFHEM/blob/5162b0bd6b4c1f262f7b0329f369a69158b9cdb9/FHEM/90_SIGNALduino_un.pm#L220

Reply:
genau richtig! So ist die aussagekräftigste Endfassung denke ich
>my $info = "$bitData; length:".length($bitData)."; hex:$rawData";	      # log example, 2018-09-19_13:43:34 SIGNALduino_unknown_84 receive: 0000001100110101010000001111110110111100; length:40; hex:033540FDBC

Reply:
Dann brauchen wir die Prüfung `if (defined($info));` nicht, oder doch?